camel-idea-plugin/src/main/java/com/github/cameltooling/idea/runner/debugger/CamelDebuggerSession.java

```diff
@@ -308,8 +317,14 @@ public void stepOver(XSourcePosition position) {
 
     public void stepOut(XSourcePosition position) {
         try {
-            XmlTag currentTag = IdeaUtils.getService().getXmlTagAt(project, position);
-            String breakpointId = getBreakpointId(currentTag);
+            Map<String, PsiElement> breakpointElement = createBreakpointElementFromPosition(position);
+
+            if (breakpointElement == null) {
+                //TODO Log warning
+                return;
+            }
+            String breakpointId = breakpointElement.keySet().iterator().next();
+
             //Get the route of the current tag
             Element routeElement = getParentRouteId(breakpointId);
             String routeId = routeElement.getAttribute("id");
```

Comment:
it is checked for nullity but not for emptiness?

If I understand well the createBreakpointElementFromPosition(position); is expected to create a Map of a single element each time? Won't it be better to provide a specific data structure for it?

Reply:
Yes, I thought about it, but it is not used anywhere else, so I decided just to use a typed singleton map. And the way it's created it cannot be empty, either a null or containing a single key-value pair.